From afaa23144d1c950e1e0b8beac9341371c5f2f2c2 Mon Sep 17 00:00:00 2001
Message-Id: <afaa23144d1c950e1e0b8beac9341371c5f2f2c2.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Tue, 27 Jan 2015 13:34:21 +0530
Subject: [PATCH 42/95] MUST_REBASE [VPG]: video/adf/intel: Enable DSI PLL for
 both DSI0 and DSI1 for dual link

For Dual link MIPI Panels, dsipll clock for both DSI0 and DSI1 needs to be enabled.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

For: GMINL-5276
Change-Id: I8ec00e595a21d5ff95cf28584801420b4669bc17
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c b/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c
index 0e901ae..0cb749b 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c
@@ -214,6 +214,7 @@ static void vlv_dsi_pll_configure(struct vlv_pll *pll)
 {
 	u32 ret;
 	struct dsi_config *intel_dsi = pll->config;
+	struct dsi_context *dsi_ctx = &intel_dsi->ctx;
 	struct dsi_mnp dsi_mnp;
 	u32 dsi_clk;
 
@@ -230,9 +231,12 @@ static void vlv_dsi_pll_configure(struct vlv_pll *pll)
 		return;
 	}
 
-	if (pll->port_id == PORT_A)
+	/* Enable DSI0 pll for DSI Port A & DSI Dual link */
+	if (dsi_ctx->ports & (1 << PORT_A))
 		dsi_mnp.dsi_pll_ctrl |= DSI_PLL_CLK_GATE_DSI0_DSIPLL;
-	else
+
+	/* Enable DSI1 pll for DSI Port C & DSI Dual link */
+	if (dsi_ctx->ports & (1 << PORT_C))
 		dsi_mnp.dsi_pll_ctrl |= DSI_PLL_CLK_GATE_DSI1_DSIPLL;
 
 	pr_info("dsi pll div %08x, ctrl %08x\n",
-- 
1.7.9.5

